### ISSUE TITLE
'SeparatorsPaint' and 'Legend' are wrong in 'Change Series Instance'

### ISSUE THREAD

**Issue by kojima-imamura** [beto-rodriguez/LiveCharts2#928]
If you use 'SeparatorsPaint' and 'Legend' in 'ChangeSeriesInstance' of the sample program, the display will collapse.
'SeparatorsPaint' is displayed in a wrong order when used with 'DrawMarginFrame'.
The display position of 'Legend' is different every time. Characters disappear.

Version in which the problem occurred
--> beta.701

https://user-images.githubusercontent.com/118875864/217446191-f4beee20-3b5c-4aaa-8a97-ac0c4ff21327.mp4

.\LiveCharts2-master\samples\WPFSample\Test\ChangeSeriesInstance\View.xaml
Edit Code
`<lvc:CartesianChart Grid.Row="1" Series="{Binding CartesianSeries}" />`
--> `<lvc:CartesianChart Grid.Row="1" Series="{Binding CartesianSeries}" DrawMarginFrame="{Binding DrawMarginFrame}" YAxes="{Binding YAxes}" LegendPosition="Bottom" />`

.\LiveCharts2-master\samples\ViewModelsSamples\Test\ChangeSeriesInstance\ViewModel.cs
Edit Code
```
    [ObservableProperty]
        private IGeoSeries[]? _geoSeries;
```
-->
```
    [ObservableProperty]
        private IGeoSeries[]? _geoSeries;

    [ObservableProperty]
        private DrawMarginFrame? _drawMarginFrame;

    [ObservableProperty]
        private Axis[]? _yAxes;
```

Edit Code
```
        GeoSeries = new IGeoSeries[]
        {
            new HeatLandSeries
            {
                Lands = new HeatLand[]
                {
                    new() { Name = "bra", Value = 13 },
                    new() { Name = "mex", Value = 10 },
                    new() { Name = "usa", Value = 15 },
                    new() { Name = "can", Value = 8 }
                }
            }
        };
```
-->
```
        GeoSeries = new IGeoSeries[]
        {
            new HeatLandSeries
            {
                Lands = new HeatLand[]
                {
                    new() { Name = "bra", Value = 13 },
                    new() { Name = "mex", Value = 10 },
                    new() { Name = "usa", Value = 15 },
                    new() { Name = "can", Value = 8 }
                }
            }
        };

        DrawMarginFrame = new DrawMarginFrame
        {
            Stroke = new SolidColorPaint(SKColors.Black, 2),
        };

        YAxes = new Axis[]
        {
            new Axis
            {
                MinLimit = 0,
                SeparatorsPaint = new SolidColorPaint
                {
                    Color = SKColors.LightGray,
                    StrokeThickness = 2
                },
            }
        };
```

**Comment by beto-rodriguez** [comment]
Thanks for the report, fixed in #1021, this will be included in the next version of the library.

**Comment by kojima-imamura** [comment]
Confirmed the fix.
thank you!
